### ISSUE TITLE
adopt a code of conduct

### ISSUE THREAD

**Issue by lsmith77** [symfony/diversity#1]
see https://github.com/symfony/symfony/pull/24896

The proposed one is likely the most popular one in the open source community and in general CoC’s are like open source licenses: you don’t roll your own.

However there are of course others that well established communities have adopted like https://www.djangoproject.com/conduct/ and http://citizencodeofconduct.org.

**Comment by lsmith77** [comment]
some useful advice plus lots of links to even more advice:
https://medium.com/@fox/when-a-code-of-conduct-becomes-harmful-1d4e737ff7aa

**Comment by lsmith77** [comment]
useful article in case of questions about CoCs
https://www.ashedryden.com/blog/codes-of-conduct-101-faq

**Comment by lsmith77** [comment]
to some extend this ticket is a bit blocked by #2 .. however I think it would be better to then simply name Fabien until we have solved #2 .. which will hopefully not take endlessly long. but better to move forward on this imho.

**Comment by egircys** [comment]
Some input from my side also:

A popular CoC is specified by the TODO Group. It is followed by the member companies. http://todogroup.org/members/

Among others
https://github.com/twitter/code-of-conduct/blob/master/code-of-conduct.md
https://code.facebook.com/pages/876921332402685/open-source-code-of-conduct

Their work on the code of conduct will not be continued, as stated in their GitHub. 
https://github.com/todogroup/opencodeofconduct

Among the suggested CoC's is the Contributor Covenant, which confirms that it is the right choice.

Also, I believe it is the right choice not to fork the existing CoC, as mentioned by @lsmith77 in this ticket and in https://github.com/symfony/symfony/pull/24896#discussion_r152219647. But, if needed we can add a Appendix to it like in https://github.com/nodejs/admin/blob/master/CODE_OF_CONDUCT.md. The enforcement group from #2 could be mentioned there.

**Comment by nicolas-grekas** [comment]
Hello, I have a few concerns I gathered about the current proposal that I would like answered before I could really take position on this. Here they are, in logic order:

1. the Covenant Coc has been proposed, many others have been proposed also on eg Slack (I linked Rust's) - but I cannot find any benchmark/comparison on our side - no report of that. This makes me uncomfortable with the claim of the (draft) blog post that ought to announce we chose this one after extensive comparison. I don't feel like that's the case (maybe just me), this is any issue IMHO because of 2.:
2. the Covenant made some uncomfortable, and I can understand because it tackles the CoC topic from the "negative"/"dark" side. My feeling hen reading it is that people behave poorly and they engage to stop doing so, and if not, a police group is put in place, with the capability to restrict my liberties (related to the project of course, not in a freedom). I understand we need an enforcement group, but I really miss a positive part to the document, that explains we are (engage to be) welcoming and benevolent. Other CoC do have that aspect (thus my concern in 1.)
3. I don't necessarily agree with Lukas on the topic we should not write our own CoC: a CoC is not a legal document like a license is. We should be able to derivate our own if we want to, like so many other did before us. I'd like this axiom to the discussion be reconsidered/challenged.
4. I had proposed a tripartite organization for conflict resolution rule. I really think we should have it from day 1 - It seems like a noticeable improvement over current policy strategies (and seemed supported on slack).

Translated with www.DeepL.com/Translator

**Comment by javiereguiluz** [comment]
My concerns are about the enforcement group:

1) I'd like the full enforcement group to be active since the very moment the CoC is officially introduced. To me it's not enough to put Fabien and Lukas names on the CoC. I need the whole group names and I hope the enforcement group is diverse, not just two males :)

2) I'd also like to know how are those members elected, reelected and expelled; also, who and how controls them.

3) I'd need full transparency of all their actions. For example, every case reported to them must be added to a public changelog. Obviously all data related to identities gets removed. Example of a report: *"A person reported that two people harassed them on Symfony Slack. We talked with person A and they admitted the wrongdoing, apologized to the victim and promised to behave in the future. Person B rejected the accusations but given the proofs in our possession, we banned them from the Symfony Slack for 3 months."*

**Comment by lsmith77** [comment]
@javiereguiluz so to summarize a bit, you prefer #2 to be solved before or together with #1.

I generally agree we need an enforcement team, that enforcement team needs processes for how to handle reports and how to handle transparency (note full transparency is likely not the right way to go) and that enforcement team needs some system for how they are appointed (here I also don't think that elections are the right way to go).

that being said, what really changes with the adoption of the CoC. all we have done is make more clear that we don't want, but fundamentally what is in the CoC already matches what we think today. As such today we do not have any of the things you mention, there is no enforcement group, there is no election and there is no process for transparency. So delaying this ticket, because #2 isn't done, does not make sense to me. Again  I totally agree, I want those processes defined and I want an enforcement team ASAP.

**Comment by fabpot** [comment]
I don't think we need to rush things out. I would really prefer to have positive initiatives for diversity first, instead of having a CoC. I agree with Nicolas when he talks about positivism.

**Comment by nicolas-grekas** [comment]
For reference, I'm the PHP RFC on the topic: https://wiki.php.net/rfc/adopt-code-of-conduct
and the withdrawal from Antony Ferrara: https://externals.io/message/90726
The withdrawal is nice to read to remind us the traps we should NOT fall into :)

**Comment by lsmith77** [comment]
Thanks for posting this @nicolas-grekas .. Anthony Ferrara perfectly sums up my sentiment of why I don't think we need to wait longer or add additional requirements before adoption.

**Comment by egircys** [comment]
Thank you everyone for moving this topic further! From my side I just can confirm what @lsmith77 already said. 

Sadly, the PHP RFC was not approved. I believe it would have made this decision of adopting the CoC much easier.

For the enforcement group, I believe it's an essential part of the CoC and should be definitely solved with #2. I'm not sure if having a enforcement team really blocks the adoption. I think the status of the diversity initiative is really transparent, and no one will judge for not having fully specified processes from the start.

**Comment by michellesanver** [comment]
I think we can only apply a code of conduct if we have defined a minimum process. Because otherwise... What do we do if we have a code of conduct and there is a violation? Where will it be reported? Who will handle the case? Is that person able to handle CoC violations... ? 

We may not need the entire process and enforcement team as in #2 but we need to have a minimal solution for these questions in the interim. A code of conduct should only exist if from the moment it exists we promise that we adhere to it and enforce it.

**Comment by lsmith77** [comment]
I have my next meeting with otter.technology next Wednesday, I will raise this topic there. Maybe Sage Sharp can advice on some minimal processes to bootstrap things.

**Comment by lsmith77** [comment]
what do you think of https://github.com/SoutheastPHPConf/policies/blob/southeastphp/reporting_guidelines.md as a starting point?

we would need to make it more international, ie. referencing contacting local police if there is thereat of (I would add “immediate” here, as in non “intermediate” situations I would recommend to first discuss if involving police is helpful or not) physical danger. Also of course we would need to adapt references to SouthEastPHP and reference out code of conduct for the contact details.

I would place these guidelines in the docs and inter-link them with the CoC going into the main repo, or what do you think would make the most sense?

**Comment by lsmith77** [comment]
One thing I wonder about the reporting guidelines document is to remind people that on Slack we only have acess to the last 10k messages, so that especially anything in public channels needs to be reported quickly to enable a 3rd party to “collect evidence”.

Also note that the above guidelines do not really cover the topic if record keeping. This is something I would suggest we look into with the enforcement team. There is a lot of complexity here due to privacy laws in Europe that need to be taken into account. The last topic that is missing is how to handle conflicts of interest. Again this is something I would look into as part of #2.

**Comment by lsmith77** [comment]
for record keeping I created a new issue https://github.com/symfony/diversity/issues/24

**Comment by lsmith77** [comment]
a few more options for processes:
* http://www.writethedocs.org/code-of-conduct-response/
* https://wealljs.org/code-of-conduct
* https://frameshiftconsultingdotcom.files.wordpress.com/2016/03/conferenceincidentresponseguide.pdf (I have shared this document before, it however is more for an internal process that isn't published)

I will try to look over all the options but I invite others to help me there to ensure we either choose the right one .. or come up with the right combination of pieces to take from each of them.

**Comment by michaelcullum** [comment]
It doesn't cover enforcement/process at all but it's worth mentioning (it's used by a few projects including ZF): http://codemanifesto.com/

Does the CoC and the enforcement information need to be in the same document?

**Comment by nicolas-grekas** [comment]
Another very detailed code of conduct, from FreeBSD: https://www.freebsd.org/internal/code-of-conduct.html

**Comment by michellesanver** [comment]
I really like the FreeBSD one, it covers cases which is often forgotten or “borderline” but important for a friendly space. 💚

Not sure I fully agree with all the enforcements but it’s difficult to come up with a better list...

**Comment by sstok** [comment]
https://github.com/symfony/symfony-docs/pull/9394 is merged now so I guess we can close this one?

**Comment by lsmith77** [comment]
w00p w00p!

**Comment by egircys** [comment]
Linking the page https://symfony.com/doc/current/contributing/code_of_conduct/index.html
